### ISSUE TITLE
GPL-nnn As a SSA (Louise Aigrain) I want the control samples picked by the Beckman robot into a destination plate to have the control batch identifier in their supplier_sample_name field so that control batches can be traced through to analysis

### ISSUE THREAD

**Issue by TWJW-SANGER** [sanger/General-Backlog-Items#89]
**User story**
As a SSA (Louise A) I want the control samples picked by the Beckman robot into a destination plate to have the control batch identifier in their supplier_sample_name field so that control batches can be traced through to analysis

**Who are the primary contacts for this story**
Nicholas R nr6@sanger.ac.uk; 
Lesley S ls7@sanger.ac.uk;
Scott T st2@sanger.ac.uk;

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Picked control samples will have the control batch id entered in their supplier_name field
- [ ] Picked control samples will be correctly identified as positive or negative control within SequenceScape and MLWH
- [ ] Cherry picking documentation updated to detail this

**Dependencies**
This story is blocked by the following dependencies:

**Additional context**
DNAP will make up batches of positive and negative control, these batches will have unique identifiers associated with them (possibly assigned from Mixtio software).
During cherry picking there will be one source control plate containing one (large) well of positive control and one (large) well of negative control.
The robot will place one control of each type on every destination plate in a random position - the resulting control sample in the destination plate should have the control batch id in the supplier_name field so failures and issues can be traced back to control batches.

**Comment by harrietc52** [comment]
GitLab psd-deployment PR 130

**Comment by andrewsparkes** [comment]
Email chain with Louise to clarify (latest at top):

> As long as we can distinguish the negative from the positive controls in the control type column, they can both have the same positive control aliquot barcode. It might actually become handy to know that these 2 were added at the same time.
> Many thanks,
> Louise
>  
> From: Andrew S
> Subject: Re: GPL736 - Control batches on the Beckmans
>  
> Should that positive control barcode be recorded against just the positive control sample in the LIMS, or also the negative?
>  
> It doesn’t sound like too hard a story, but I think it unlikely to be ready for version 1 as we’ve still a fair bit to do on the current functionality.
> 
> On 23 Nov 2020, at 11:43, Louise A wrote:
> Hi,
>  
> Sorry the only barcode that will go on the control plate will be the positive control unique aliquot barcode since we can just assume that the negative is water (single use too). So no issue with the number of label per plate.
> However talking with Scott, we definitively don’t want it hardcoded within the Beckman’s software.
> This positive control barcode is indeed the one we’d like to see saved in the LIMS. Mixteo should then allow us to link it to an original batch.
>  
> Currently Sm is printing and using a same barcode for every control plates which is obvious not ideal but we could continue doing so in the interim until this story is in place.
>  
> Many thanks,
> Louise
>  
> From: Andrew S
> Sent: 23 November 2020 11:25
> Subject: Re: GPL736 - Control batches on the Beckmans
>  
> Thanks for the quick response Louise,
>  
> So what’s on the robot is a single plate (not 2 troughs) containing both controls, but with 2 barcodes on it? That’s going to be tricky, plates normally have only one identifier.
> Is it obvious which is the positive control barcode and which is the negative to the user?
>  
> And it’s these barcode numbers you want stored in the supllier sample name for the controls in the LIMS yes?
>  
> I think Beckman may have difficulty auto-scanning 2 barcodes from one plate (presumably on different sides of the plate?). So we'll have to consider how those numbers are going to be entered into the Beckman method.
>  
> Andrew
>  
> On 23 Nov 2020, at 11:10, Louise A wrote:
>  
> Hi Andrew,
>  
> 1.       A batch of each control is made (maybe in Mixtio) and each batch get a unique identifier BUT ALSO several barcodes are created for each aliquots of the batch (that where Mixteo should be useful as it’s good at linking aliquots to original batches).
> 2.       The 2 unique single-use aliquots’ barcodes are printed and stuck to the side pf the plate containing the positive and negative controls on the Beckman deck (so not transfer between Beckman or anything like that)
> Lesley, Scott and Nick, please correct me if there is anything inaccurate in my description.
>  
> Also Lesley and Scott, what do we think about the hardcoding the barcode in the robot’s method?
>  
> Many thanks,
> Louise
>  
> From: Andrew S
> Sent: 23 November 2020 11:02
> Subject: GPL736 - Control batches on the Beckmans
>  
> Hi Louise,
>  
> Can we clarify our understanding is correct:
> You make a ‘batch’ of the control (positive or negative) (maybe in Mixtio)
> The batch gets a unique identifier
> That identifer is printed off as a barcode label and stuck on the side of the trough put on the Beckman?
> Multiple Beckmans may be using troughs with the same batch barcode on them?
>  
> If that’s the case:
> How are you expecting the batch barcode to get into the Beckman method so that it can be passed through into the LIMS?
> The robot cannot pick up a trough to scan it presumably (liquid would slop everywhere).
> Beckman are talking about hardcoding those batch identifiers in their method, is that acceptable? You’d have to remember to edit the methods on all the robots if they changed.
>  
> NB. we plan to also record which of the 4 Beckman robots source plates and destination plate events are used on (via their serial numbers), so you should be able to trace back if there are systematic sequencing issues.

**Comment by rl15** [comment]
Can we use Gatekeeper?

**Comment by JamesGlover** [comment]
We do have https://github.com/sanger/gatekeeper/issues/100 open on Gatekeeper, and this would be using functionality which it otherwise currently unused. 

- Gatekeeper *does* allow the creation of reporter plates (plates of samples, not tags), however these would use a fixed plate layout, so would have the same positive control sample for all lots.
- These plates would then exist in Sequencescape as fully realised plates, however I guess there would still be a requirement to somehow get them into Dart?
- In addition, as the pick happens outside of Sequencescape, we wouldn't be able to map the samples back to their source well, without updating the output plate creation to explicitly do so (such as with a transfer request)
- In the absence of a custom sample per lot, reporting would rely on extracting the original reporter plate from Seqeuncescape. This is quite convoluted, as there is no direct association between the original well, and the aliquot in the Sequencing lane. Walking the request graph is brittle and slow.

It might be that there are some shortcuts I'm overlooking to help address the last point.

**Comment by TWJW-SANGER** [comment]
A little more info: Louise & Danni have had a demo of [Mixtio](https://mixtio.psd.sanger.ac.uk) with Laura 2/12/2020 and I believe this can work for their lab workflow without disrupting CGaP's use of the system.

**Comment by andrewsparkes** [comment]
A little on how it works on Beckman now:
The control transfers are marked as positive or negative in a 'control' well attribute in DART by the Beckman method. All other wells on the plate have this field too, but are empty (null or empty string). The control plate barcode and well are also recorded as well attributes, 'source_barcode' and 'source_coordinate'. The Beckman randomises the location of the 2 controls on the plate. The lighthouse API endpoint is called on completion of the plate to create the destination in Sequencescape. Controls are created as control samples, and show up visually in Limber as +/- in the plate.

**Comment by emrojo** [comment]
Control plate barcodes can be extracted from the control samples subjects created for the plate creation event, but it would be better for reporting to have the control plate barcodes as a new subject in the plate creation event.
I'm adding a new acceptance criteria, to also track the control plates as subjects in the plate creation event as it would be better for reporting.

**Comment by Skrich1999** [comment]
This is no longer needed as there are work arounds in place and Louise has left the institute.  A new acceptance criteria was added 13 months ago, to also track the control plates as subjects in the plate creation event as it would be better for reporting.  If this new criteria were needed a new story should be created and prioritized
